pennylane/ops/math/utils.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=too-few-public-methods,function-redefined
+
+import pennylane as qml
+
+
+def flatten_decomposition(decomp):
+    """Unpack nested lists of operator products."""
+    if not decomp:  # empty list
+        return decomp
+    if isinstance(decomp[0], qml.ops.math.MatMul):
+        return flatten_decomposition(
+            [decomp[0].hyperparameters["left"], decomp[0].hyperparameters["right"]]
+        ) + flatten_decomposition(decomp[1:])
+    return decomp[:1] + flatten_decomposition(decomp[1:])
+
+
+def flatten_terms(coeffs, ops):
```

Comment:
Logic is exactly as above, but more messy because we extract two objects...could make this function take the tuple of two objects!